src/rgw/rgw_lua_background.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+#include "common/dout.h"
+#include "rgw_common.h"
+#include <string>
+#include "rgw_lua_utils.h"
+
+namespace rgw::lua {
+
+//Interval between each execution of the script is set to 5 seconds
+constexpr const int INIT_EXECUTE_INTERVAL = 5;
+
+//Writeable meta table named RGW with mutex protection
+using BackgroundMap = std::unordered_map<std::string, std::string>;
+struct RGWTable : StringMapMetaTable<BackgroundMap,
+  StringMapWriteableNewIndex<BackgroundMap>> {
+    static std::string TableName() {return "RGW";}
+    static std::string Name() {return TableName() + "Meta";}
+    static int IndexClosure(lua_State* L) {
+      const auto mtx = reinterpret_cast<std::mutex*>(lua_touserdata(L, lua_upvalueindex(2)));
+      std::lock_guard l(*mtx);
+      return StringMapMetaTable::IndexClosure(L);
+    }
+    static int LenClosure(lua_State* L) {
+      static const auto mtx = reinterpret_cast<std::mutex*>(lua_touserdata(L, lua_upvalueindex(2)));
+      std::lock_guard l(*mtx);
+      return StringMapMetaTable::LenClosure(L);
+    }
+    static int NewIndexClosure(lua_State* L) {
+      static const auto mtx = reinterpret_cast<std::mutex*>(lua_touserdata(L, lua_upvalueindex(2)));
+      std::lock_guard l(*mtx);
+      return StringMapMetaTable::NewIndexClosure(L);
+    }
+};
+
+class Background {
+
+private:
+  BackgroundMap rgw_map;
+  std::string rgw_script;
+  bool stopped = false;
+  int execute_interval = INIT_EXECUTE_INTERVAL;
+  const DoutPrefixProvider* dpp;
+  rgw::sal::Store* store;
+  std::thread runner;
+  CephContext* cct;
+  std::string luarocks_path;
+  std::mutex m_mutex;
+
+  void run();
+
+public:
+  Background(const DoutPrefixProvider* dpp,
+      rgw::sal::Store* const store,
```

Comment:
```suggestion
      rgw::sal::Store* store,
```
there is not harm, but this is not needed, all arguments are local vars, so we assume we don't change them
to clarify, this is like declaring:
```
void foo(const int i) {
...
}
```